### ISSUE TITLE
Voice fine-tuning jobs live in a Map that is never evicted, so checkpoints become unpromotable after a restart

### ISSUE THREAD

**Issue by atomantic** [atomantic/PortOS#5664]
**Problem** — `activeJobs` is the only record of a Qwen3-TTS fine-tuning run. Nothing ever deletes from it, and nothing ever persists it. Two consequences: (1) every run leaves a `jobState` (including its full `checkpoints` array and a dead `AbortController`) resident for the life of the process — the only in-memory job registry in the codebase without the `closeJobAfterDelay(jobs, jobId, …)` eviction every sibling uses; (2) `getFineTuningJobStatus` and `promoteCheckpoint` both look the job up by id and throw `JOB_NOT_FOUND` (404) when it is missing, so after any PortOS restart the operator can no longer promote a checkpoint that is sitting complete on disk under `<profileDir>/fine-tune/<jobId>/`. Training runs for tens of minutes, which makes "PortOS restarted while it ran / shortly after it finished" the common case, not the edge case. The child process is also untracked outside this map, so shutdown has no handle on it.

**Evidence** — `server/services/voice/fineTuning.js:29-30` and `:128`:

    // In-memory active training job map
    const activeJobs = new Map();
    …
    activeJobs.set(jobId, jobState);

There is no `activeJobs.delete` / `activeJobs.clear` anywhere in the file, and no `closeJobAfterDelay` import (every peer registry — `server/services/videoDownload.js`, `server/services/youtubeIngest.js`, `server/services/musicVideo/render.js`, `server/services/imageGen/agy.js`, `server/services/loraDatasetCaption.js` — calls it). The three consumers all 404 on a miss, `server/services/voice/fineTuning.js:209-214`, `:221-226`, `:236-241`:

    export function getFineTuningJobStatus(jobId) {
      const job = activeJobs.get(jobId);
      if (!job) { throw new ServerError('Fine-tuning job not found', { status: 404, code: 'JOB_NOT_FOUND' }); }
    …
    export async function promoteCheckpoint({ profileId, jobId, checkpointId }) {
      const job = activeJobs.get(jobId);
      if (!job) { throw new ServerError('Fine-tuning job not found', { status: 404, code: 'JOB_NOT_FOUND' }); }
      const ckpt = job.checkpoints.find((c) => c.id === checkpointId || String(c.step) === String(checkpointId));

and the terminal handlers (`child.on('close')` at `:180`, `child.on('error')` at `:194`) mutate `jobState` but never touch the map.

**Plan**
1. Persist the job record next to its artifacts. In `startFineTuning`, after `await mkdir(outputDir, { recursive: true })`, write a `job.json` in `outputDir` with the serializable slice of `jobState` (everything except `controller`), using `atomicWrite` from `server/lib/fileUtils.js`. Re-write it from `child.on('close')` and `child.on('error')` so the terminal `status`/`error`/`completedAt` and the final `checkpoints` array land on disk. Decision: a sidecar `job.json` under the profile's `fine-tune/<jobId>/` directory, not a new store — the checkpoints it indexes already live there, so the record and its artifacts are deleted together.
2. Add `async function loadJob(jobId, profileId)` that returns `activeJobs.get(jobId)` when present and otherwise reads `<profileArtifactDirectory(profileId)>/fine-tune/<jobId>/job.json`. Route `getFineTuningJobStatus` and `promoteCheckpoint` through it (both already receive or can derive `profileId` — `promoteCheckpoint` takes it as a parameter; give `getFineTuningJobStatus` a `profileId` argument and thread it from `server/routes/voice.js`). Keep the `JOB_NOT_FOUND` 404 for a job with neither.
3. Evict the in-memory entry once it is terminal and on disk: in both terminal handlers, after the `job.json` write, call `closeJobAfterDelay(activeJobs, jobId)` from `server/lib/sseUtils.js` (import it) so a status poll racing completion still hits memory for the grace window and the map does not grow.
4. Track the child for shutdown: store `child` on `jobState`, and in `cancelFineTuningJob` keep the existing `controller.abort()` (the spawn already carries `signal`), adding nothing new — this step is only the assignment so a future shutdown hook has a handle.

**Tests** — extend `server/services/voice/fineTuning.test.js`. Pin: (a) after a fake child exits 0, `job.json` exists in `outputDir` with `status: 'completed'` and the recorded checkpoints — uniquely catches the missing persistence; (b) with `activeJobs` cleared (simulating a restart), `promoteCheckpoint` still resolves the checkpoint from `job.json` instead of 404ing — uniquely catches the unpromotable-after-restart bug; (c) the in-memory entry is gone after the eviction delay.

**Acceptance criteria**
- [ ] A completed fine-tuning run writes `job.json` alongside its checkpoints.
- [ ] `promoteCheckpoint` works for a job whose in-memory entry is gone.
- [ ] `activeJobs` is empty once every run is terminal plus the SSE grace window.
- [ ] `cd server && npm test -- fineTuning` passes.

**Out of scope** — do not add a shutdown hook that kills in-flight training children, and do not change the checkpoint-promotion semantics in `promoteFineTunedProfile`, in the same PR.

---
Filed by a `/do:better --scan-only --issues` audit (2026-09-01). Category: `bugs-perf` · Severity: `low` · Files: server/services/voice/fineTuning.js:30, server/services/voice/fineTuning.js:128, server/services/voice/fineTuning.js:210, server/services/voice/fineTuning.js:222, server/services/voice/fineTuning.js:236

All labels already exist in the repo; do NOT create labels. Never add `planner:*` labels.
